src/string_mutable.zig

```diff
@@ -427,6 +428,33 @@ pub const MutableString = struct {
         try self.list.appendSlice(self.allocator, bytes);
         return bytes.len;
     }
+
+    /// Reads a file into this MutableString.
+    /// An `over_alloc` of 0 will be increased to 1.
+    /// Guarantees `list.items.len = file_size`, and `list.capacity > list.items.len + over_alloc`
+    pub fn readFile(
+        self: *MutableString,
+        file: std.fs.File,
+        /// The number of bytes which should be allocated at the end of the buffer after the file contents.
+        over_alloc: usize,
+    ) !void {
+        const overhead = @max(1, over_alloc);
+
+        // (fast path) Try reading into the buffer without checking how big the file is
+        const already_read = try file.read(self.list.allocatedSlice());
+        self.list.items.len = already_read;
+        if (self.list.capacity >= try std.math.add(usize, already_read, overhead)) return;
```

Comment:
On most OS's, `already_read` can fit in a `u32`. If we wanted, we could eliminate the overflow for most OS's by making the `over_alloc` a `u32` and listing the OS's that only allow reading `std.math.maxInt(u32)` bytes (or less) at once.